tools/dedupe-types.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileoverview Strips typedef aliases from the rolled-up file. This
+ * is necessary because the TypeScript compiler throws an error when
+ * it encounters a duplicate typedef.
```

Comment:
Do you have a reference here? My instinct is that one of the following must be true:

* This is a case of the repo's tooling "holding it wrong": which would be a good thing to fix (e.g. by switching from Rollup to something like tsup that's more commonly used with TS builds)
* This is an actual bug in TypeScript that we can link to & I'd want to know more about
* This is a really common thing we'd want to have standardized tooling for (which would surprise me, but you never know!)

Reply:
I thought this was intended behavior? Basically, if you have two JSDoc typedefs with the same name, `tsc` errors out. You can test this for yourself by checking out this branch and removing the dedupe-types step from `npm run build` and then running it.

Reply:
It might be intended behavior on the TypeScript side to error on this case, but it's not intended behavior for the case to be hit in the first place. In other words: it's not intended that everybody writing types-in-JS should have to write tools to work around quirks of type checking and transpiling in this way. This kind of bespoke tooling shouldn't be normal or required for any kind of TypeScript projects.

Reply:
Gotcha. I guess I assumed it was akin to defining the same variable twice with `let` or `const`.